python/Ganga/new_tests/GPI/newXMLTest/TestXMLGenAndLoad.py

```diff
@@ -0,0 +1,159 @@
+from __future__ import absolute_import
+
+from ..GangaUnitTest import GangaUnitTest
+
+from os import path, stat
+
+import time
+
+def getJobsPath():
+    from Ganga.Runtime.Repository_runtime import getLocalRoot
+    jobs_path = path.join(getLocalRoot(), '6.0', 'jobs')
+    return jobs_path
+
+def getXMLDir(this_job):
+    jobs_path = getJobsPath()
+    _id = this_job.id
+    jobs_master_path = path.join(jobs_path, "%sxxx" % str(int(_id/1000)))
+    return path.join(jobs_master_path, str(_id))
+
+def getXMLFile(this_job):
+    return path.join(getXMLDir(this_job), 'data')
+
+def getIndexFile(this_job):
+    return path.join(getXMLDir(this_job), '../%s.index' % str(this_job.id))
+
+testStr = "testFooString"
+
+class TestXMLGenAndLoad(GangaUnitTest):
+
+    def setUp(self):
+        """Make sure that the Job object isn't destroyed between tests"""
+        super(TestXMLGenAndLoad, self).setUp()
+        from Ganga.Utility.Config import setConfigOption
+        setConfigOption('TestingFramework', 'AutoCleanup', 'False')
+
+    def test_a_JobConstruction(self):
+        """ First construct the Job object (singular)"""
+        from Ganga.Utility.Config import getConfig
+        self.assertFalse(getConfig('TestingFramework')['AutoCleanup'])
+
+        from Ganga.GPI import Job, jobs
+        j=Job()
+        self.assertEqual(len(jobs), 1) # Don't really gain anything from assertEqual...
```

Comment:
It's copy/paste from a legacy test, I'm not expecting these tests to be fully debugged yet, this is a bit WIP but I was interested in feedback.